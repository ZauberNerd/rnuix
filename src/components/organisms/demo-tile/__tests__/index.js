// @flow
jest.unmock('../');

import * as React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

import DemoTile from '../';

const shallow = new ShallowRenderer();
const render = jest.fn();

describe('DemoTile', () => {
    it('renders default', () => {
        const tree = shallow.render(<DemoTile title="Foo" render={render} />);
        expect(tree).toMatchSnapshot();
    });

    it('renders fullscreen', () => {
        const tree = shallow.render(
            <DemoTile title="Foo" render={render} isFullScreen />,
        );
        expect(tree).toMatchSnapshot();
    });

    it('calls the passed render function', () => {
        const tree = renderer.create(<DemoTile title="Foo" render={render} />);
        expect(render).toHaveBeenCalled();
    });
});
