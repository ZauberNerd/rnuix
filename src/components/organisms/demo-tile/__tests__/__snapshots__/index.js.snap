// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DemoTile renders default 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 3,
        "shadowOffset": Object {
          "height": 1.2,
        },
        "shadowOpacity": 0.183,
        "shadowRadius": 1.08,
      },
      undefined,
    ]
  }
>
  <DemoHeader
    onEnterFullScreen={undefined}
    title="Foo"
  />
  <DemoRenderer
    isFullScreen={false}
    render={[Function]}
  />
</View>
`;

exports[`DemoTile renders fullscreen 1`] = `
<View
  style={
    Array [
      null,
      undefined,
    ]
  }
>
  <DemoRenderer
    isFullScreen={true}
    render={[Function]}
  />
</View>
`;
